test/e2e/aws/sts/actutator_e2e_test.go

```diff
@@ -4,13 +4,132 @@
 
 import (
+	"context"
+	minterv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"github.com/openshift/cloud-credential-operator/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
+	"os"
+	"sigs.k8s.io/e2e-framework/klient/conf"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/envfuncs"
+	"sigs.k8s.io/e2e-framework/pkg/features"
 	"testing"
+	"time"
 )
 
-// Test_CheckSTS runs end-to-end tests to verify that the results of an STS workflow are successful.
+var testenv env.Environment
+var secret = &corev1.Secret{}
 
-func Test_CheckSTS(t *testing.T) {
-	t.Run("test of STS", func(t *testing.T) {
-		t.Parallel()
-	})
+const (
+	name       = "test-sts-creds-req"
+	namespace  = "default"
+	secretName = "test-sts-secret"
+)
+
+func TestMain(m *testing.M) {
+	testenv = env.New()
+	path := conf.ResolveKubeConfigFile()
+	cfg := envconf.NewWithKubeConfig(path)
+	testenv = env.NewWithConfig(cfg)
+	namespace := envconf.RandomName("sample-ns", 16)
+	testenv.Setup(
+		envfuncs.CreateNamespace(namespace),
+	)
+	testenv.Finish(
+		envfuncs.DeleteNamespace(namespace),
+	)
+	os.Exit(testenv.Run(m))
+}
+
+func TestSecretCreationOnCredsRequestWithSTSInfo(t *testing.T) {
+	t.Logf("About to create a CredentialsRequest for testing Secret creation")
+	cr := newCredentialsRequest()
+	credReqFeature := features.New("minterv1.CredentialsRequest").
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// create a CredentialsRequest
+			client := cfg.Client()
+			util.SetupScheme(scheme.Scheme)
+			if err := client.Resources().Create(ctx, cr); err != nil {
+				t.Fatal(err)
+			}
+			// watch for the Secret and trigger action based on the event received.
+			client.Resources().Watch(
+				&corev1.SecretList{},
+				resources.WithFieldSelector(
+					labels.FormatLabels(map[string]string{"metadata.name": secretName}))).
+				WithAddFunc(onAdd(t, *cfg, ctx)).
+				Start(ctx)
+			return ctx
+		}).
+		Assess("secret creation", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			time.Sleep(2 * time.Minute)
+			if secretName != secret.GetName() {
+				t.Errorf("Secret name is incorrect. Expected: `%s`", secretName)
+			}
+			return context.WithValue(ctx, "test-secret", secret)
+		}).
+		Teardown(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			if err := cfg.Client().Resources().Delete(ctx, cr); err != nil {
+				t.Fatal(err)
+			}
+			return ctx
+		}).Feature()
+	testenv.Test(t, credReqFeature)
+}
+
+func onAdd(t *testing.T, cfg envconf.Config, ctx context.Context) func(obj interface{}) {
+	time.Sleep(2 * time.Minute)
+	if err := cfg.Client().Resources().Get(ctx, secretName, namespace, secret); err != nil {
```

Comment:
This is bit due to unsure about Secret really existing here, despite this being a callback from the watch on the Secret.

Reply:
The `obj interface{}` passed to this func is the object that was just added - perhaps use that? 2min wait should not be needed here